vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java

```diff
@@ -34,17 +39,36 @@
 
     @Autowired
     private SequenceRepository sequenceRepo;
-    
+
+
+    @Autowired
+    private BranchingPointRepository bpointRepo;
+
+    @Autowired
+    private ChoiceRepository choiceRepo;
+
     private final Logger logger = LoggerFactory.getLogger(getClass());
-    
-    @Override
-    public ISlide createSlide(String moduleId, SlideForm slideForm) {
-        IModule module = moduleManager.getModule(moduleId);
-        ISlide slide = slideFactory.createSlide(module, slideForm);
-        slideRepo.save((Slide) slide);
-        return slide;
+
+    @Override
+    public ISlide createSlide(IModule module, SlideForm slideForm, SlideType type) {
+        ISlide slide = slideFactory.createSlide(module, slideForm, type);             
+        slide = slideRepo.save((Slide) slide);        
+        return slide;
+    }
+
+    @Override
+    public IBranchingPoint createBranchingPoint(IModule module, SlideForm slideForm, SlideType type) {
+        ISlide branchingPoint = slideFactory.createSlide(module, slideForm, type);            
+        branchingPoint = bpointRepo.save((BranchingPoint) branchingPoint);        
+        return (IBranchingPoint) branchingPoint;
     }
 
+    @Override
+    public IBranchingPoint updateBranchingPoint(IBranchingPoint branchingPoint) {
+        branchingPoint = bpointRepo.save((BranchingPoint) branchingPoint);        
+        return (IBranchingPoint) branchingPoint;

```

Comment:
merge with line 68 or don't return anything (when updating you already have an id)